### PR TITLE
Remove cursor state from redux store

### PR DESCRIPTION
**Pull request by pcardune**
We were doing a poor job of keeping codemirror's cursor and redux's cursor in sync, and in turns out that we don't actually use the redux cursor in any place where we don't already have access to codemirror itself, so duplicating the state really wasn't necessary.

This PR removes all that code.

### CODE REVIEW

File: packages/codemirror-blocks/src/ui/BlockEditor.tsx [hunk 1]
```diff
@@ -167,30 +166,6 @@ const BlockEditor = ({ options = {}, ...props }: BlockEditorProps) => {
...
-        const { cur } = getState();
-        if (cur != null) {
-          return; // if we already have a cursor, bail
-        }
-        const node = focusedNode || ast.getFirstRootNode();
-        dispatch(
-          activateByNid(editor, node && node.nid, {
-            allowMove: true,
-          })
-        );
```
Comment (pcardune) [inline packages/codemirror-blocks/src/ui/BlockEditor.tsx:189]: This was the only place where we were doing something with the redux cursor that was different than the behavior of codemirror's cursor. Our redux version of the cursor was allowed to be `null`, but codemirror's cursor is always a something. I tried all kinds of things to get the application into a state where the cursor was actually null, and couldn't figure out how to do it. When I removed this function altogether, all the tests still passed, which makes me think this function wasn't doing anything useful to begin with.

**Review by pcardune — COMMENTED** [review]
Aside from `handleTopLevelFocus()`, which I got rid of completely, the only other place where we were using the cursor from the redux store was in the keyboard event handlers for navigation between nodes. But they already had access to the codemirror instance (or at least our wrapper around it), and could just use codemirror's version of the cursor.

**Comment by schanzer** [comment]
Here's the use case that's a problem, which I've previous complained is very difficult to test:

Switch to block mode, then hit Tab repeatedly. The focus order should be: (1) search functions input gets focus, then (2) `42` gets focus. In this branch, `42` never receives focus. Further use of the tab key makes some weird stuff happen!

**Comment by pcardune** [comment]
Thanks for the info, I will look into this.

**Comment by pcardune** [comment]
ok, I see the problem with that particular case, but I'm not sure I understand the whole picture of the intended behavior. If I start with focus on the search box, and then hit tab, am I *always* supposed to end up with focus on the first node? Because that's not what happens on master if you follow these steps:

- Fresh page load
- Press `TAB` (end up on search box)
- Press `TAB` (end up on first node)
- Pres `CTRL+[` (end up with a cursor before the first node)
- Click on the search box
- Press `TAB` (end up with a cursor before the first node again)

I suspect whatever the behavior we're after can be accomplished without storing a copy of the cursor.

**Comment by schanzer** [comment]
@pcardune the behavior on master is correct. The block editor itself, from a screenreader perspective is an "application". That means it's responsible for managing its own focus state (among other things), and that shifting focus off and back onto the application should always return focus to the place it was to begin with.

The code in question was responsible for one thing only: determining the behavior on the _first_ focus. When the application state has no focus defined, it should default to the first root (if it exists) or `{line:0, ch:0}` (if it doesn't).

**Comment by pcardune** [comment]
Ok, I've restored `handleTopLevelFocus` using local state (in a ref unfortunately) instead of global state since this is only used by BlockEditor for something very very specific.

**Comment by schanzer** [comment]
Looks great!

**Pull request merged**